server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -1070,19 +1070,127 @@ public <C extends NamedWriteable> C readOptionalNamedWriteable(Class<C> category
      *
      * @return the list of objects
      * @throws IOException if an I/O exception occurs reading the list
+     * @deprecated use {@link #readCollectionAsList}.
      */
+    @Deprecated(forRemoval = true)
```

Comment:
Do these need to stay here for external uses (plugins)?

Reply:
This is just to avoid breaking the serverless build, I'll remove them once I've fixed that up.

Reply:
Gotcha